### PR TITLE
[Select] Fix alignment issues

### LINKED ISSUE #1357: [Select] Alignment is broken after upgrading to React 18

### PR DESCRIPTION
**Pull request by benoitgrelard**
Closes #1357

### CODE REVIEW

**Comment by benoitgrelard** [comment]
Green!!! ✅

File: packages/react/select/src/Select.tsx [hunk 1]
```diff
@@ -1031,7 +1042,7 @@ const SelectScrollUpButton = React.forwardRef<
   const [canScrollUp, setCanScrollUp] = React.useState(false);
   const composedRefs = useComposedRefs(forwardedRef, contentContext.onScrollButtonChange);
 
-  React.useEffect(() => {
+  useLayoutEffect(() => {
```
Comment (benoitgrelard) [inline packages/react/select/src/Select.tsx:1045]: I believe these were the only thing needed to make it go green and the reason I saw the jumps and misalignments in React 18.

File: packages/react/select/src/Select.tsx [hunk 2]
```diff
@@ -475,20 +480,25 @@ const SelectContentImpl = React.forwardRef<SelectContentImplElement, SelectConte
         const willAlignWithoutTopOverflow = contentTopToItemMiddle <= topEdgeToTriggerMiddle;
 
         if (willAlignWithoutTopOverflow) {
+          const isLastItem = selectedItem === items[items.length - 1].ref.current;
           contentWrapper.style.bottom = 0 + 'px';
+          const scrollClearance = isLastItem ? viewportPaddingBottom : 0;
           const viewportOffsetBottom =
-            content.clientHeight - viewport.offsetTop - viewport.offsetHeight;
+            content.clientHeight - viewport.offsetTop - viewport.offsetHeight + scrollClearance;
```
[the remark this answers is not in the document]
Comment (benoitgrelard) [reply packages/react/select/src/Select.tsx:487]: Also, not super happy with the new variable names (`scrollClearance` and the now overloaded `viewportOffsetBottom`).

Perhaps, I shouldn't bundle them as now it's not truly `viewportOffsetBottom`…

[the remark this answers is not in the document]
Comment (benoitgrelard) [reply packages/react/select/src/Select.tsx:487]: Unfortunately no, we'd have to do single stories for it to be caught, let's catchup on a call.

File: .storybook/main.js [hunk 3]
```diff
@@ -4,8 +4,6 @@ module.exports = {
   stories: ['../packages/react/**/*.stories.tsx', '../packages/core/**/*.stories.tsx'],
   addons: ['@storybook/addon-storysource'],
 
-  reactOptions: { legacyRootApi: true },
```
Comment (andy-hook) [inline .storybook/main.js:7]: Are we planning to enable `stictMode` separately?

**Review by andy-hook — APPROVED** [review]

File: .storybook/main.js (hunk 3 above)
Comment (andy-hook) [reply .storybook/main.js:7]: Spoke too soon 😀

**Pull request merged**